watchtower/wtclient/client.go

```diff
@@ -423,10 +427,10 @@ func (c *TowerClient) buildHighestCommitHeights() {
 
 		// Take the heights commit height found in the session's acked
 		// updates.
-		for _, bid := range s.AckedUpdates {
-			height, ok := chanCommitHeights[bid.ChanID]
-			if !ok || bid.CommitHeight > height {
-				chanCommitHeights[bid.ChanID] = bid.CommitHeight
+		for chanID, commitHeight := range s.MaxChanCommitHeights {
```

Comment:
This is the only time we need access to `s.MaxChanCommitHeights`. For nodes with lots of sessions it might be a good idea to remove this information from memory as soon as possible, instead of waiting for `buildHighestCommitHeights` to finish.

Reply:
see latest update :) hopefully solves this issue!